modules/4337/test/erc4337/ERC4337ModuleExisting.spec.ts

```diff
@@ -99,15 +115,20 @@ describe('Safe4337Module - Existing Safe', () => {
       const safeOpHash = calculateSafeOperationHash(await validator.getAddress(), safeOp, await chainId())
       const signature = buildSignatureBytes([await signHash(user1, safeOpHash)])
       const userOp = buildUserOperationFromSafeUserOperation({ safeOp, signature })
-      await logGas('Execute UserOp with fee payment', entryPoint.executeUserOp(userOp, ethers.parseEther('0.000001')))
-      expect(await ethers.provider.getBalance(await safe.getAddress())).to.be.eq(ethers.parseEther('0.499999'))
+      await logGas('Execute UserOp with fee payment', entryPoint.handleOps([userOp], randomAddress))
+
+      // checking that the fee was paid
+      expect(await ethers.provider.getBalance(randomAddress)).to.be.gt(0)
+      // check that the call was executed
+      expect(await ethers.provider.getBalance(randomAddress2)).to.be.eq(ethers.parseEther('0.5'))
     })
 
     it('reverts on failure', async () => {
       const { user1, safe, validator, entryPoint } = await setupTests()
 
-      await user1.sendTransaction({ to: await safe.getAddress(), value: ethers.parseEther('0.000001') })
-      expect(await ethers.provider.getBalance(await safe.getAddress())).to.be.eq(ethers.parseEther('0.000001'))
+      // Make sure to send enough ETH for the pre-fund but less than the transaction
+      await user1.sendTransaction({ to: await safe.getAddress(), value: ethers.parseEther('0.4') })
+      expect(await ethers.provider.getBalance(await safe.getAddress())).to.be.eq(ethers.parseEther('0.4'))
```

Comment:
You can prefund the account using `depositTo` to the EntryPoint contract, which would (IMO) make this test closer to the original and easier to understand.